qtaf-core/src/main/java/de/qytera/qtaf/core/log/model/collection/TestScenarioLogCollection.java

```diff
@@ -526,6 +526,20 @@ public synchronized <T extends LogMessage> List<T> getLogMessages(Class<T> logMe
                 .toList();
     }
 
+    /**
+     * Get the log message object of the step that is currently pending
+     * @return  Step log object of the currently pending step
+     */
+    public synchronized StepInformationLogMessage getStepLogOfPendingStep() {
+        List<LogMessage> logMessages = LogMessageIndex.getInstance().getByScenarioIdAndPending(getScenarioId());
+
+        if (logMessages.size() > 0) {
+            return (StepInformationLogMessage) logMessages.get(0);
+        }
+
+        return null;
```

Comment:
With the changes below, this can be simplified to:

```java

        List<StepInformationLogMessage> pendingSteps = LogMessageIndex.getInstance().getByScenarioIdAndPending(getScenarioId());
        return pendingSteps.isEmpty() ? null : pendingSteps.get(0);
```

Reply:
Refactored the method the way it was suggested